tsl/src/continuous_aggs/refresh.c

```diff
@@ -565,7 +565,9 @@ continuous_agg_refresh(PG_FUNCTION_ARGS)
 													  get_fn_expr_argtype(fcinfo->flinfo, 1),
 													  refresh_window.type);
 	else
-		refresh_window.start = ts_time_get_min(refresh_window.type);
+		refresh_window.start = ts_continuous_agg_bucket_width_variable(cagg) ?
+								   ts_time_get_nobegin_or_min(refresh_window.type) :
+								   ts_time_get_min(refresh_window.type);
 
```

Comment:
Don't you need to fix the calls to `ts_time_get_max` below in a similar manner?

Reply:
I've just aligned this user API call to our internal logic when we create a new cagg don't passing `WITH NO DATA` in the view definition to also refresh the entire cagg: https://github.com/timescale/timescaledb/blob/main/tsl/src/continuous_aggs/create.c#L2929-L2945